### ISSUE TITLE
Local Function Emulator Triggering wrong function

### ISSUE THREAD

**Issue by markdagraca** [firebase/firebase-tools#4189]
Trying to run my react typescript project using Local Emulator and it is triggering the wrong function when called from the front end. Code is working perfectly in the production and test environment but not in Local

**Comment by markdagraca** [comment]
Downgraded to 10.1.5 and it appears to be working properly but I'm now missing env variables features found in 10.2

**Comment by Albertbol** [comment]
exactly the same issue for the node js environment, all latest libraries
```
    "engines": {
        "node": "16"
    },
    "main": "dist/index.js",
    "dependencies": {
        "firebase": "^9.6.7",
        "firebase-admin": "^10.0.2",
        "firebase-functions": "^3.18.1"
    },
```

**Comment by roobd** [comment]
I have the same issue that my function triggers are not working anymore after the upgrade to 10.2.1.
There is only one onCreate function triggered all the time, even a creation on a completely different document path is triggering this one function.

A downgrade to 10.2.0. does not fix the issue.

I use firebase-functions at version 3.18.1

A downgrade to 10.1.5 is fixing the issue and all triggers are working as expected again.

**Comment by taeold** [comment]
We are taking a look at the issue. Sorry for the incovenience!

**Comment by ggirotto** [comment]
We were facing the same issue but it stopped happening yesterday.

**Comment by gustavrannestig** [comment]
I have the same issue. From what I can see the patterin is that it seems to be stuck on the first called function.

So when I restart the emulators, I can choose what function to use for that "session" by firing that first. But then I can't switch and call another function without restarting.

So I'm guessing some sort of cache bug? :)

**Comment by haroldadmin** [comment]
Can confirm, facing the same issue that @gustavrannestig described. It keeps triggering the first triggered function for any subsequent API calls.

**Comment by taeold** [comment]
Been investigating the issue, and @gustavrannestig's description helps a ton. Will post an update here once I get a sense of the bug - apologies to everyone for the inconvenience.

**Comment by Zebreus** [comment]
I also have the same issue. The first triggered function is called for any subsequent API calls.
The first bad firebase-tools version is 10.2.0.
The issue only happens, when the functions emulator is in debug mode.

**Comment by taeold** [comment]
@Zebreus Thanks for a very helpful pointer.

Is everyone reporting the issue also starting the emulator with `--inspect-functions` flag?

**Comment by markdagraca** [comment]
Yes im using the following command to launch the tool firebase
emulators:start --inspect-functions --export-on-exit=./saved-data
--import=./saved-data

**Comment by markdagraca** [comment]
What @Zebreus said is exactly what I was experiencing

**Comment by gustavrannestig** [comment]
@taeold yes I'm also running the emulators with `--inspect-functions`

**Comment by nikgavalas** [comment]
I removed the `--inspect-functions` flag and everything seems to work just fine. If I add it in I can confirm that the issue occurs again. Hopefully that helps in tracking down the fix!

**Comment by doo-gl** [comment]
Similar issue, starting the emulator, it will repeatedly run whatever was the initial function regardless of what path is used.

Downgrading to 10.1.5 resolves the issue for now.

For those looking for what to do, run:
```
npm install -g firebase-tools@10.1.5
```
check your running version is 10.1.5 using 
```
firebase --version
```
If it is still 10.2.x, restart your terminal.

**Comment by taeold** [comment]
Hi folks. The patch for this issue was released today in firebase-tools v10.2.2. Please give it a spin and let us know if things are working expected.
